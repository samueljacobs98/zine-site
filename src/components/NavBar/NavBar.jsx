import "./NavBar.scss";

const NavBar = () => {
  return <div>NavBar</div>;
};

export default NavBar;
