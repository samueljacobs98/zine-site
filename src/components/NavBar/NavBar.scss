@use "../../styles/variables.scss" as *;
.NavBar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  background-color: $primary-color;
}
.NavBarDesktop {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  margin: 0 2rem;
  background-color: $primary-color;
  &__icons {
    display: flex;
    gap: 1rem;
  }
}
