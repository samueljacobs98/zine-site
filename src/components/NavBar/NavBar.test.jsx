import { customRender } from "../../utils/testUtils";
import NavBar from "./NavBar.jsx";

it("should match NavBar component snapshot", () => {
  const { container } = customRender(<NavBar />);

  expect(container).toMatchSnapshot();
});

it("should match window width resize", () => {
  Object.defineProperty(window, "innerWidth", {
    writable: true,
    configurable: true,
    value: 768,
  });

  window.dispatchEvent(new Event("resize"));

  expect(window.innerWidth).toBe(768);
});
