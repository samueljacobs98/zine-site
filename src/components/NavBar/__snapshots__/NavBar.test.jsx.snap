// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match NavBar component snapshot 1`] = `
<div>
  <div
    class="NavBarDesktop"
  >
    <img
      alt=""
      src="Menu.svg"
    />
    <h2
      class="NavBarDesktop__title"
    >
      ZINESTORE
    </h2>
    <div
      class="NavBarDesktop__icons"
    >
      <img
        alt="Search Icon"
        src="Search.svg"
      />
      <img
        alt="Upload Icon"
        src="Upload.svg"
      />
      <img
        alt="Notifications Icon"
        src="Notifications.svg"
      />
    </div>
  </div>
</div>
`;
